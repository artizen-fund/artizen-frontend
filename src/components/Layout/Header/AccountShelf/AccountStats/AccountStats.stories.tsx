import AccountStats from './'

export default {
  title: 'components/AccountStats',
  component: AccountStats,
  argTypes: {},
}

export const AccountStatsComponent = (props: any) => {
  const mockedStats = [
    {
      glyph: 'donate',
      unit: '$800',
      label: 'donated',
    },
    {
      glyph: 'info',
      unit: '42',
      label: 'collected',
    },
    {
      glyph: 'info',
      unit: '$0',
      label: 'collected',
    },
    {
      glyph: 'info',
      unit: '100',
      label: '$ART',
    },
  ]
  return <AccountStats {...props} stats={mockedStats} />
}
