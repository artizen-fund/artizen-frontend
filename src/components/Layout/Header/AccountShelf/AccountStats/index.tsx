import styled from 'styled-components'
import { Icon } from '@components'
import { typography, palette } from '@theme'
import { rgba } from '@lib'

interface IAccountStats {
  stats: Array<{
    glyph: string
    unit: string
    label: string
  }>
}

const AccountStats = ({ stats }: IAccountStats) => {
  return (
    <Stats>
      {stats.map((stat, i) => (
        <Stat key={`stat-${i}`}>
          <Icon glyph={stat.glyph} outline />
          <Unit>{stat.unit}</Unit>
          <Label>{stat.label}</Label>
        </Stat>
      ))}
    </Stats>
  )
}

const Stats = styled.ul`
  grid-area: stats;

  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(2, 1fr);

  li:nth-of-type(1),
  li:nth-of-type(3) {
    border-right: 0.5px solid ${rgba(palette.stone)};
    @media (prefers-color-scheme: dark) {
      border-right: 0.5px solid ${rgba(palette.barracuda, 0.64)};
    }
  }

  li:nth-of-type(1),
  li:nth-of-type(2) {
    border-bottom: 0.5px solid ${rgba(palette.stone)};
    @media (prefers-color-scheme: dark) {
      border-bottom: 0.5px solid ${rgba(palette.barracuda, 0.64)};
    }
  }
`

const Stat = styled.li`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

const Unit = styled.div`
  ${typography.title.l4}
  color: ${rgba(palette.night)};
  margin-top: 0.5em;
  @media (prefers-color-scheme: dark) {
    color: ${rgba(palette.moon)};
  }
`

const Label = styled.div`
  ${typography.label.l2}
  color: ${rgba(palette.barracuda)};
`

export default AccountStats
