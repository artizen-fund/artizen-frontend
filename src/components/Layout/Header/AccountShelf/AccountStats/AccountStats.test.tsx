import renderer from 'react-test-renderer'
import AccountStats from './'

describe('AccountStats', () => {
  const mockedStats = [
    {
      glyph: 'donate',
      unit: '$800',
      label: 'donated',
    },
    {
      glyph: 'info',
      unit: '42',
      label: 'collected',
    },
    {
      glyph: 'info',
      unit: '$0',
      label: 'collected',
    },
    {
      glyph: 'info',
      unit: '100',
      label: '$ART',
    },
  ]
  it('renders correctly', () => {
    const tree = renderer.create(<AccountStats stats={mockedStats} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
