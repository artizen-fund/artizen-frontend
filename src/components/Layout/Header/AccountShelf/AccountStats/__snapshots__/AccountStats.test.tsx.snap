// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountStats renders correctly 1`] = `
<ul
  className="sc-cf63bf03-0 jnGrA-D"
>
  <li
    className="sc-cf63bf03-1 doApLo"
  >
    <div
      className="sc-60e75bb7-2 cdqnon"
    >
      <div
        className="sc-60e75bb7-0 kBbFeK"
        color="night"
      >
        <div
          className="sc-149d3190-0 ewzPBY"
          color="night"
        />
      </div>
    </div>
    <div
      className="sc-cf63bf03-2 kOxwXQ"
    >
      $800
    </div>
    <div
      className="sc-cf63bf03-3 kdPNii"
    >
      donated
    </div>
  </li>
  <li
    className="sc-cf63bf03-1 doApLo"
  >
    <div
      className="sc-60e75bb7-2 cdqnon"
    >
      <div
        className="sc-60e75bb7-0 kBbFeK"
        color="night"
      >
        <div
          className="sc-149d3190-0 cyhlIT"
          color="night"
        />
      </div>
    </div>
    <div
      className="sc-cf63bf03-2 kOxwXQ"
    >
      42
    </div>
    <div
      className="sc-cf63bf03-3 kdPNii"
    >
      collected
    </div>
  </li>
  <li
    className="sc-cf63bf03-1 doApLo"
  >
    <div
      className="sc-60e75bb7-2 cdqnon"
    >
      <div
        className="sc-60e75bb7-0 kBbFeK"
        color="night"
      >
        <div
          className="sc-149d3190-0 cyhlIT"
          color="night"
        />
      </div>
    </div>
    <div
      className="sc-cf63bf03-2 kOxwXQ"
    >
      $0
    </div>
    <div
      className="sc-cf63bf03-3 kdPNii"
    >
      collected
    </div>
  </li>
  <li
    className="sc-cf63bf03-1 doApLo"
  >
    <div
      className="sc-60e75bb7-2 cdqnon"
    >
      <div
        className="sc-60e75bb7-0 kBbFeK"
        color="night"
      >
        <div
          className="sc-149d3190-0 cyhlIT"
          color="night"
        />
      </div>
    </div>
    <div
      className="sc-cf63bf03-2 kOxwXQ"
    >
      100
    </div>
    <div
      className="sc-cf63bf03-3 kdPNii"
    >
      $ART
    </div>
  </li>
</ul>
`;
